### PR TITLE
 WebXRManager: support more than 2 views

### PR DESCRIPTION
**Pull request by CodyJasonBennett**
Related issue: #19387

**Description**

I was chatting with @BryanChrisBrown about WebXRManager's limitation to two XRCameras (also noted in https://github.com/three-types/three-ts-types/pull/169) and found @brianpeiris's previous work sufficient enough to support multiview as described in https://github.com/kainino0x/holoplay-webxr/issues/9#issuecomment-955620666.

I tested this with a few demos and a (webxr) emulator equipped with multiview to try:

<details>
  <summary>r140 dev (baseline)</summary>

Demo: https://webxr-multicam-4zetdnsmg-codyb.vercel.app/
Code: https://github.com/CodyJasonBennett/webxr-multicam/blob/r140dev-baseline/index.html

[image](https://webxr-multicam-4zetdnsmg-codyb.vercel.app/)

</details>

<details>
  <summary>fix/webxr-multicamera (PR)</summary>

Demo: https://webxr-multicam.vercel.app/
Code: https://github.com/CodyJasonBennett/webxr-multicam/blob/main/index.html

[image](https://webxr-multicam.vercel.app/)

</details>

The only notable changes since have been with RGB depreciations, internal encoding changes, and foveation & reference space features so this worked OOTB.

### CODE REVIEW

**Comment by cabanier** [comment]
Awesome work!
Does this also work if the scene has reflections?

File: src/renderers/webxr/WebXRManager.js [hunk 1]
```diff
@@ -621,6 +630,8 @@ class WebXRManager extends EventDispatcher {
 
 					}
 
+					if ( ! cameras[ i ] ) addCamera( i );
+
 					const camera = cameras[ i ];
```
Comment (mrdoob) [inline src/renderers/webxr/WebXRManager.js:636]: This is how we usually do it:

```js
let camera = cameras[ i ];

if ( camera === undefined ) {

	camera = new PerspectiveCamera();
	camera.layers.enable( i );
	camera.viewport = new Vector4();
	cameras[ i ] = camera;

}
```

**Comment by CodyJasonBennett** [comment]
@cabanier, I haven't noticed anything wrong with rendering. Is there a specific example I can look at?

**Comment by cabanier** [comment]
> @cabanier, I haven't noticed anything wrong with rendering. Is there a specific example I can look at?

try https://threejs.org/examples/?q=vr#webxr_vr_sandbox

**Comment by CodyJasonBennett** [comment]
I'd have to ask @BryanChrisBrown for what this would look like on a real device, but looks fine with multiple views.

Demo: https://webxr-multicam-2v9u5dip5-codyb.vercel.app
Code: https://github.com/CodyJasonBennett/webxr-multicam/blob/reflections/index.html

image

**Comment by CodyJasonBennett** [comment]
I have a [branch with a build](https://github.com/CodyJasonBennett/three.js/tree/webxr-multicamera-build) to test without an emulator: https://rawcdn.githack.com/CodyJasonBennett/three.js/d31e899eebf6a9282668672abd49e9bcd6ce6f07/examples/index.html

**Comment by CodyJasonBennett** [comment]
I suppose see #24048 for an implementation.

**Comment by CodyJasonBennett** [comment]
Are there any withstanding issues with this PR? AFAIK OCULUS_multiview (see #24048) would be limited to headsets like the Quest, but it looks like the code path for other WebXR devices would remain intact.

**Comment by mrdoob** [comment]
All good. Thanks!

**Pull request merged**